lisa/runners/lisa_runner.py

```diff
@@ -635,6 +636,10 @@ def _merge_test_requirements(
                         original_node_requirement = schema.load_by_type(
                             schema.NodeSpace, node_requirement_data
                         )
+                        original_node_requirement.disk = node_requirement.disk
```

Comment:
Why are the two lines needed?

Reply:
This was essentially the issue. `original_node_requirement` when loaded by type converts from AzureDiskSettings to DiskSettings, removing the field `has_resource_disk`. This is fixing the issue by reassigning the correct value.

Reply:
When it's loaded from type, the unsupported parts should be in extended schemas. And it will be loaded back when pass the prepare environment of platform.